src/apps/explorer/components/SummaryCardsWidget/VolumeChart.tsx

```diff
@@ -0,0 +1,215 @@
+}
+
+function _buildChart(
+  chartContainer: HTMLDivElement,
+  width: number | undefined,
+  height: number,
+  theme: DefaultTheme,
+): IChartApi {
+  return createChart(chartContainer, {
+    width,
+    height,
+    handleScroll: false,
+    handleScale: false,
+    layout: {
+      backgroundColor: 'transparent',
+      textColor: theme.textPrimary1,
+    },
+    rightPriceScale: {
+      scaleMargins: {
+        top: 0.5,
+        bottom: 0.2,
+      },
+      visible: false,
+    },
+    leftPriceScale: {
+      visible: false,
+    },
+    timeScale: {
+      visible: false,
+    },
+    grid: {
+      horzLines: {
+        visible: false,
+      },
+      vertLines: {
+        visible: false,
+      },
+    },
+    crosshair: {
+      horzLine: {
+        visible: false,
+        labelVisible: true,
+      },
+      vertLine: {
+        visible: true,
+        style: 3,
+        width: 1,
+        color: theme.borderPrimary,
+        labelVisible: true,
+      },
+    },
+  })
+}
+
+export function VolumeChart({
+  volumeData,
+  height = DEFAULT_CHART_HEIGHT,
+  width = undefined,
+  periodId = DEFAULT_PERIOD_ID,
+  children,
+}: React.PropsWithChildren<VolumeChartProps>): JSX.Element {
+  const { data: items, currentVolume, changedVolume, isLoading } = volumeData || {}
+  const chartContainerRef = useRef<HTMLDivElement>(null)
+  const [chartCreated, setChartCreated] = useState<IChartApi | null | undefined>()
+  const theme = useTheme()
+  const diffPercentageVolume = currentVolume && changedVolume && calcDiff(currentVolume, changedVolume)
+  const captionNameColor = getColorBySign(diffPercentageVolume || 0)
+  const [crossHairData, setCrossHairData] = useState<HistogramData | null>(null)
+  const previousPeriod = usePreviousLastValueData(periodId)
+
+  // reset the chart when the volume period is changed
+  useEffect(() => {
+    if (periodId !== previousPeriod && chartCreated) {
+      chartCreated.resize(0, 0)
+      setChartCreated(null)
+    }
+  }, [chartCreated, periodId, previousPeriod])
+
+  useEffect(() => {
+    if (chartCreated || !chartContainerRef.current || !items || isLoading) return
+
+    const chart = _buildChart(chartContainerRef.current, width, height, theme)
+    const series = chart.addAreaSeries({
+      lineWidth: 1,
+      lineColor: theme[captionNameColor],
+      topColor: theme[captionNameColor],
+      bottomColor: captionNameColor === 'red1' ? COLOR_NEGATIVE_DIFFERENCE : COLOR_POSITIVE_DIFFERENCE,
+    })
+
+    series.setData(items)
+
+    chart.subscribeCrosshairMove(function (param: MouseEventParams) {
+      if (param === undefined || param.time === undefined || !param.point || param.point.x < 0 || param.point.y < 0) {
+        setCrossHairData(null)
+        return
+      }
+
+      const value = param.seriesPrices.get(series) as BarPrice
+      const time = param.time
+      setCrossHairData({ time, value })
+    })
+
+    chart.timeScale().fitContent()
+    if (!isLoading) setChartCreated(chart)
```

Comment:
This condition will always be true, otherwise it would return on line 148, no?

Reply:
I forgot to delete it while testing the loading.